### PR TITLE
Add task to generate docs

### PR DESCRIPTION
**Pull request by lbalmaceda**
### Description

Added an npm script to generate the docs into a `docs` folder in the root.

### References
Used [Typedoc](https://typedoc.org/) to achieve this. Ideally, the docs generation will be another step in the release pipeline.

### Testing

- [ ] This change adds test coverage for new/changed/fixed functionality

### Checklist

- [x] I have added documentation for new/changed functionality in this PR or in auth0.com/docs
- [x] All active GitHub checks for tests, formatting, and security are passing
- [x] The correct base branch is being used, if not `master`

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -8,6 +8,7 @@
     "test": "ng test",
     "test-ci": "ng test --no-watch --no-progress --browsers=ChromeHeadlessCI --codeCoverage=true",
     "lint": "ng lint",
+    "docs": "typedoc",
```
Comment (stevehobbsdev) [inline package.json:13]: This is great - let's also execute this whenever we build, I suggest putting it into a `postbuild` script:

```
"postbuild": "npm run docs"
```

Comment (stevehobbsdev) [reply package.json:13]: Also, let's make sure we publish a link to the docs in the readme, see [Auth0 React](https://github.com/auth0/auth0-react/#documentation).

Comment (stevehobbsdev) [reply package.json:13]: One other thing - when you create the `postbuild` script, can you also refactor the `npm run useragent` command from the `build` script into a new `prebuild` script?

So:

```json
"build": "npm run update-useragent && ng build",
```

would become:

```json
"prebuid": "npm run update-useragent",
"build": "ng build"
```

**Review by stevehobbsdev — CHANGES_REQUESTED** [review]
Left a couple more things, rejecting just so you see the changes.

File: package.json (hunk 1 above)
Comment (lbalmaceda) [reply package.json:13]: Done 🏁

Comment (stevehobbsdev) [reply package.json:13]: Awesome! And thanks for referencing the quickstart too, nice job 👏

**Review by stevehobbsdev — APPROVED** [review]

**Pull request merged**